src/hooks/use-api-data.ts

```diff
@@ -0,0 +1,60 @@
+import useSWR from 'swr';
+import { siteConfig } from '@/site.config';
+
+interface FetcherOptions {
+    url: string,
+    method?: 'GET' | 'POST',
+    params?: Record<string, string>,
+    // body?: Record<string, any>
```

Comment:
@qafui is this intentional to be commented out?

Reply:
Yes. I realized that this hook is a little too generic. I suggest we split it into a few custom hooks (one for credentialed `POST` fetches and another for simple `GET` requests where we can use more `Cache` and revalidation config options to change the behavior of the `useSWR` hook. We would need that for taking notes and more live streaming behaviors.